prow/cmd/genjobs/genjobs.go

```diff
@@ -0,0 +1,382 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	flag "github.com/spf13/pflag"
+	"sigs.k8s.io/yaml"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	prowjob "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+)
+
+const (
+	autogenHeader     = "# THIS FILE IS AUTOGENERATED. DO NOT EDIT. See prow/cmd/genjobs/README.md\n"
+	modifier          = "private"
+	filenameSeparator = "."
+	jobnameSeparator  = "_"
+	yamlExt           = ".(yml|yaml)$"
+)
+
+// Command-line flags
+
+var bucket string
+var cluster string
+var clean bool
+var sshKeySecret string
+var labels map[string]string
+var branches []string
+var input string
+var output string
+var repoWhitelist sets.String
+var repoBlacklist sets.String
+var jobWhitelist sets.String
+var jobBlacklist sets.String
+var orgMap map[string]string
+
+// validateOrgRepo validates that the org and repo for a job pass validation and should be converted.
+func validateOrgRepo(org string, repo string) bool {
+	_, hasOrg := orgMap[org]
+
+	if !hasOrg || repoBlacklist.Has(repo) || (len(repoWhitelist) > 0 && !repoWhitelist.Has(repo)) {
+		return false
+	}
+
+	return true
+}
+
+// validateJob validates that the job passes validation and should be converted.
+func validateJob(name string, patterns []string) bool {
+	if jobBlacklist.Has(name) || (len(jobWhitelist) > 0 && !jobWhitelist.Has(name)) || !isMatchBranch(patterns) {
+		return false
+	}
+
+	return true
+}
+
+// validateFlags validates that the command-line flags.
+func validateFlags() {
+	var err error
+
+	if len(orgMap) == 0 {
+		printErrAndExit("-m, --map option is required.", 1)
```

Comment:
'mapping'? (Or is the README incorrect?)

Reply:
Yes, it should be `--mapping`. 🤦‍♂